### DISCUSSION TITLE
Potential improvements

### DISCUSSION THREAD

**Discussion by ABAG603** [community/community#11604] [Actions]
Hi,

I've been experimenting with Actions for few days already looking how it compares to Azure DevOps Pipelines that I'm using atm. I really liked workflow syntax - it's looking more refined and easy than the one Pipelines are using. Runners look to be faster than Agents as well. However, I've found that Actions are still missing few things:

- Secrets: there is no way to store variables that are not secrets. I know I still can store them as a secrets but I'd like to be able to see their values in the UI as well as in build log. Having same concept of Library and Variable Groups like Pipelines has would be very helpful.
- There is an option to re-run all jobs but no option to re-run only failed jobs.
- There is a Wait timer exists for Environment but it would be great to have a Timeout as well, so if deployment wasn't approved during the certain time period it will be cancelled. The reason is that sometimes you want only to build but not to deploy. Cancelled deployment job would not set whole pipeline as failed as well.
- Another functionality missing is to be able either to update build number or add labels to the build from steps. For example, let's say one of my jobs builds docker container - I'd like to update build number with the tag or add a label with tag, so I can easily identify which build run produced this container.

Hopefully, some of the items above are already in the roadmap.

**Comment by Omzig** [comment]
Confirmed, Azure lets you do variables that are clear text. Git Hub needs to catch up with Azure features quickly ;)

**Comment by ethomson** [comment]
> There is an option to re-run all jobs but no option to re-run only failed jobs.

We've added this functionality. 🎉 

For the other pieces, this is all great feedback, and things that we're looking to improve in the future. Thanks much for taking the time to give us the feedback.

**Reply by Omzig** [reply]
Welp, i just found out there is no option to "re-reun successful jobs", in azure you are able to manually fail a successful job, and redeploy/rerun it. 

Use case, let say during deployment the staging server gets windows updates and messes something up, who knows right? Then we click the green deployment job and then click redeploy, then it goes back through the approval process and lets you rerun the successful job.
